library/Fission/Types.hs

```diff
@@ -394,37 +396,50 @@ instance User.Creator Fission where
           Left err -> Error.relaxedLeft err
           Right _  -> Right userId
 
+runUpdate ::
```

Comment:
Okay 2 questions that I thought of here:

Does this function belong in a different module? Or is it fine here? This file is all interfaces so it seems a bit weird to have this function floating around in here, but in another file, I think we'll have to deal with a circular dependency

Should I try to include make this function work for `setData` as well? It'll definitely add some complexity since it's not quite as similar as the other 2 functions were. It uses `MonadDNSLink` instead of `MonadRoute53` and passes different parameters. I ended up not rewriting cause I thought it was more straightforward this way

Reply:
> This file is all interfaces so it seems a bit weird to have this function floating around in here

Yup, it's totally fine to have in here! It's not getting exported, so it's like a `private` function in OO. It's specific to the stuff in this module. 

> we'll have to deal with a circular dependency

Yup. You really don't have a choice. I also got stuck with some of this over in `JWT.hs` because it's a recursive structure.

> Should I try to include make this function work for `setData` as well?

I wouldn't worry about it right now. We can always break these functions into subfunctions for the common parts later, but this stuff is all so concrete and internal that I'm not too worried.

> I ended up not rewriting cause I thought it was more straightforward this way

Good call!

## Aside

All of the class instances for `Fission` need to live either here, or where the class is defined. Because we're going to split the repo into subprojects shortly, we don't want the classes to have too much dependency on `Fission`. The dependencies ideally flow the other way.

Reply:
Great that all sounds good! :+1: